### PR TITLE
Added Basic Conversation API Functionality

### PR DESCRIPTION
**Pull request by mjabali**

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by dylburger** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

File: components/symbl_ai/actions/get-topics/get-topics.mjs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      optional: true,
+      default: false,
+    },
+  },
+  async run({ $ }) {
+    try {
+      const response = await this.symblAIApp.getTopics({
+        $,
+        conversationId: this.conversationId,
+        params: `sentiment=${this.sentiment}&parentRefs=${this.parentRefs}`,
```
Comment (js07) [inline components/symbl_ai/actions/get-topics/get-topics.mjs:36]: As mentioned above, you could pass the query parameters to the `axios` call using axios request config's `params` property:
```js
      const response = await this.symblAIApp.getTopics({
        $,
        conversationId: this.conversationId,
        params: {
          sentiment: this.sentiment,
          parentRefs: this.parentRefs,
        },
      });
```

In `symbl_ai.app.mjs`:
```js
    async getTopics({
      $,
      conversationId,
      params,
    }) {
      return this.makeRequest({
        $,
        path: `/conversations/${conversationId}/topics`,
        params,
      });
    },
```

File: components/symbl_ai/actions/get-action-items/get-action-items.mjs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    },
+  },
+  async run({ $ }) {
+    try {
+      const response = await this.symblAIApp.getActionItems({
+        $,
+        conversationId: this.conversationId,
+      });
+      $.export("$summary", "Action Items successfully retrieved from the conversation");
+      return response;
```
Comment (js07) [inline components/symbl_ai/actions/get-action-items/get-action-items.mjs:24]: **Suggestion** (for this action and the other new actions)

The return value from a "list" action is typically an array of the items being listed, to simplify using the results from actions in workflows. Some examples in existing components: [Airtable](https://github.com/PipedreamHQ/pipedream/blob/78b07a8bc24adb1724467987e85743fa7bed711c/components/airtable/actions/common-list.js#L54), [Google Sheets](https://github.com/PipedreamHQ/pipedream/blob/78b07a8bc24adb1724467987e85743fa7bed711c/components/google_sheets/actions/list-worksheets/list-worksheets.js#L30), [Eventbrite](https://github.com/PipedreamHQ/pipedream/blob/78b07a8bc24adb1724467987e85743fa7bed711c/components/eventbrite/actions/get-event-attendees/get-event-attendees.js#L41)

Currently the array of `actionsItems` is nested in the return value (`{"$return_value":{"actionItems":[]}}`).

```suggestion
      const { actionItems } = await this.symblAIApp.getActionItems({
        $,
        conversationId: this.conversationId,
      });
      $.export("$summary", "Action Items successfully retrieved from the conversation");
      return actionItems;
```

File: components/symbl_ai/actions/get-action-items/get-action-items.mjs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      description: "The Id of the Conversation",
+    },
+  },
+  async run({ $ }) {
+    try {
+      const response = await this.symblAIApp.getActionItems({
+        $,
+        conversationId: this.conversationId,
+      });
+      $.export("$summary", "Action Items successfully retrieved from the conversation");
```
Comment (js07) [inline components/symbl_ai/actions/get-action-items/get-action-items.mjs:23]: **Suggestion** (for this action and the other new actions):
Include the count of items retrieved in the [`$summary`](https://pipedream.com/docs/components/guidelines/#use-summary-to-summarize-what-happened).

```suggestion
      const { actionItems } = await this.symblAIApp.getActionItems({
        $,
        conversationId: this.conversationId,
      });
      $.export("$summary", `Successfully retrieved ${actionItems.length} Action Item${actionItems.length === 1
        ? ""
        : "s"} from the conversation`);
```

**Review by js07 — COMMENTED** [review]
@mjabali Thanks again for the PRs! This looks great overall. I just made a few minor suggestions.

File: components/symbl_ai/actions/get-speech-to-text/get-speech-to-text.mjs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      optional: true,
+      default: false,
+    },
+  },
+  async run({ $ }) {
+    try {
+      const response = await this.symblAIApp.getSpeechToText({
+        $,
+        conversationId: this.conversationId,
+        params: `verbose=${this.verbose}&sentiment=${this.sentiment}`,
```
[the remark this answers is not in the document]
Comment (mjabali) [reply components/symbl_ai/actions/get-speech-to-text/get-speech-to-text.mjs:36]: Fixed in bff25f90

File: components/symbl_ai/actions/get-action-items/get-action-items.mjs (hunk 2 above)
Comment (mjabali) [reply components/symbl_ai/actions/get-action-items/get-action-items.mjs:24]: Fixed in abb1fb0d

File: components/symbl_ai/actions/get-action-items/get-action-items.mjs (hunk 3 above)
Comment (mjabali) [reply components/symbl_ai/actions/get-action-items/get-action-items.mjs:23]: Fixed in abb1fb0d

File: components/symbl_ai/actions/get-action-items/get-action-items.mjs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  description: "Get a list of all the action items generated from the conversation. See the doc [here](https://docs.symbl.ai/docs/conversation-api/action-items)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    symblAIApp,
+    conversationId: {
+      type: "string",
+      label: "Conversation Id",
+      description: "The Id of the Conversation",
+    },
```
[the remark this answers is not in the document]
Comment (mjabali) [reply components/symbl_ai/actions/get-action-items/get-action-items.mjs:15]: Fixed in f36a51a9

**Review by michelle0927 — APPROVED** [review]
Looks good to me!

File: components/symbl_ai/actions/get-job-status/get-job-status.mjs [hunk 6]
```diff
@@ -4,24 +4,22 @@ export default {
...
-  version: "0.0.1",
+  version: "0.0.2",
   type: "action",
   props: {
     symblAIApp,
     jobId: {
       type: "string",
       label: "Job Id",
       description: "The Id of the job request",
-      optional: false,
```
[the remark this answers is not in the document]
Comment (mjabali) [reply components/symbl_ai/actions/get-job-status/get-job-status.mjs:15]: That's correct @michelle0927. Symbl doesn't currently have a GET /jobs API.
Any chance we can merge the approved PRs today or tomorrow?

**Comment by dannyroosevelt** [comment]
Merge when ready, @michelle0927 !

**Pull request merged**